src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -589,6 +608,32 @@ public sealed interface MemorySegment permits AbstractMemorySegmentImpl {
      */
     long byteSize();
 
+    /**
+     * {@return the <em>maximum</em> byte alignment (which is equal to or higher than
+     * the requested byte alignment during native segment allocation)}
+     * <p>
+     * The returned alignment is always an even power of two and is derived from:
+     * <ul>
+     *     <li>Heap:
+     *     the segment offset and backing array type.</li>
```

Comment:
Note that "segment offset" is `segment.address()`. So I don't think we want to use a new term here. We can also say, more succinctly:
```
derived from the segment address <link to address()> and the type of the backing heap storage <link to heapObject> (for heap segments).
```